examples/routing_service/udp_socket_adapter/README.md

```diff
@@ -0,0 +1,165 @@
+# Example Code: Routing Service C++11 Socket Adapter
+
+## Example Description
+
+This example shows how to implement a custom `Routing Service Adapter plugin`,
+build it into a shared library and load it with Routing Service.
+
+In this concrete example, we show how to write a simple adapter in C++11 to
+receive data from a UDP socket using RTI Routing Service.
+
+The code in this directory provides the following components:
+
+-   `src/SocketAdapter` implements the plugin that is loaded by *RTI Routing
+Service*. It is responsible to create and delete connections.
+-   `src/SocketConnection` implements a connection. This component is
+responsible of the creation and deletion of `StreamReaders`.
+-   `src/SocketInputDiscoveryStreamReader` implements the logic necessary to
+propagate information about the discovered input streams (in this case
+sockets) to the Routing Service.
+-   `src/SocketStreamReader` implements an `StreamReader` that reads sample
+information from a UDP socket.
+-   `test/send_shape_to_socket.py` implements a simple tester to send shape
+type data to a UDP socket.
+
+For more details, please refer to the *RTI Routing Service SDK* documentation.
+
+## Building C++ example
+
+In order to build this example, you need to define the variables
+`CONNEXTDDS_DIR` and `CONNEXTDDS_ARCH`. You can do so by exporting them
+manually, by sourcing the `rtisetenv` script for your architecture, or by
+passing them to the `cmake` command as arguments:
+
+```bash
+mkdir build
+cd build
+cmake -DCONNEXTDDS_DIR=<Connext DDS Directory> \     # If not exported
+      -DCONNEXTDDS_ARCH=<Connext DDS Architecture> \ # If not exported
+      -DBUILD_SHARED_LIBS=ON|OFF \                   # ON is preferred
+      -DCMAKE_BUILD_TYPE=Debug|Release ..
+cmake --build .
+cd ..
+export LD_LIBRARY_PATH=build/:$LD_LIBRARY_PATH
```

Comment:
I guess you need this after building, to run right? But we're already specifying that below.

Reply:
that's true, I'll move it below